src/kagglehub/auth.py

```diff
@@ -1,2 +1,16 @@
+import logging
+
+from kagglehub.config import set_kaggle_credentials
+
+logger = logging.getLogger(__name__)
+
+
 def login():
-    raise NotImplementedError()
+    """Prompt the user for their Kaggle username and API key and save them globally."""
+
+    username = input("Enter your Kaggle username: ")
+    api_key = input("Enter your Kaggle API key: ")
+
+    set_kaggle_credentials(username=username, api_key=api_key)
+
+    logger.info("You are now logged in to Kaggle Hub.")
```

Comment:
We don't validate that the credentials are valid here. It will fail later when a user makes a download / upload call if the credentials are invalid.

Given that, this message is a bit misleading.

I would replace with:

```
logger.info("Kaggle credentials set.")
```

I think it would be great though to validate the credentials at this step. Can you file a bug, assign it to you, add the [kagglehub hotlist](https://b.corp.google.com/hotlists/5354034) and tackle this in a separate PR. 

Thank you